src/components/App.tsx

```diff
@@ -74,34 +83,36 @@ export class App extends Component<{}, State> {
   }
 
   getFilteredCards = () => {
-    const {notes, filter} = this.state;
+    //const {notes} = this.state;
+    const {notes, filter} = this.props;
 
     return notes.filter((note) => {
       return note.title.toUpperCase().includes(filter.toUpperCase());
     });
   }
 
-  handleFilterChange = (event: any) => {
+  handleFilterChange = (event: ChangeEvent<HTMLInputElement>) => {
     const {value} = event.target;
-    this.setState({filter: value});
+    this.props.dispatch(setFilter(value));
   }
 
   handleNewButtonClick = () => {
-    this.setState({modalOpen: true});
+    this.props.dispatch(openAddModal());
   }
 
   handleOnModalClose = () => {
-    this.setState({modalOpen: false});
+    this.props.dispatch(closeAddModal());
   }
 
   handleOnModalSave = (note: Note) => {
 
-    let newState = {...this.state, modalOpen: false};
-
-    if (note.title !== '') {
-      newState.notes = [...newState.notes, note];
-    }
-
-    this.setState(newState);
+    this.props.dispatch(addCard(note.id, note.title, note.text));
+    this.handleOnModalClose();
```

Comment:
Is it a good idea call dispatch(addCard) and dispatch(closeAddModal) in sequence or the right way is dispatch an unique action to do both?